idaes/core/solvers/petsc.py

```diff
@@ -517,10 +519,31 @@ def petsc_dae_by_time_element(
         between = pyo.Set(initialize=sorted(between))
         between.construct()
 
+    # Make sure that time is a ContinuousSet that has been discretized
+    if isinstance(time, pyodae.ContinuousSet):
+        try:
+            time.get_discretization_info()["nfe"]
+        except KeyError:
+            raise RuntimeError("The ContinuousSet time has not been discretized")
```

Comment:
Would this work here?
```python
if not time.get_discretization_info():
    raise RuntimeError(f"ContinuousSet {time.name} is not discretized")
```
@blnicho How would you recommend checking whether a ContinuousSet is discretized?

Reply:
I think before the set is discretized, the dictionary already exists, but is empty.

Reply:
I do this check in `pyomo.dae` using the following:

    if 'scheme' not in time.get_discretization_info():
        print("time is not discretized")